### PR TITLE
Setup terraform

### PR DESCRIPTION
**Pull request by xswoodsx**
## Trello card
[Ticket](https://trello.com/c/pCiVWSEH)

### Context
We want to set up the pipelines to 
 - Create an instance of the app when a PR is opened
 - Destroy the PR instance when it is merged in to main
 - Create an instance of the app when a PR is merged to main
 
## Changes proposed in this pull request
Adding various workflows to do the above
Adding terraform files and workspace variables for review and dev environments. 

## Guidance to review
Bits and pieces of the files have been taken from other dfe-digital repositories, i've tried my best to remove any unnecessary code. We know the deploying a PR is working however we've not tested the destroy or deploy to dev. These might need further action when it's on the main.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # https://hub.docker.com/_/microsoft-dotnet
 FROM mcr.microsoft.com/dotnet/sdk:5.0.102-ca-patch-buster-slim AS build-env
```
Comment (jkempster34) [inline Dockerfile:3]: Trying to re-wrap my head around what this does, but is this build configuration needed?

**Review by jkempster34 — APPROVED** [review]
Nice!

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 
+ARG BUILD_CONFIGURATION=Debug
+ENV ASPNETCORE_URLS=http://+:8080
```
Comment (jkempster34) [inline Dockerfile:4]: Trying to re-wrap my head around what this does, but is this build configuration needed?

Comment (xswoodsx) [reply Dockerfile:4]: I don't think we do. I'll remove it

**Review by young-rob — APPROVED** [review]
LGTM!

**Pull request merged**